src/hotspot/share/opto/macro.cpp

```diff
@@ -370,30 +370,30 @@ Node *PhaseMacroExpand::value_from_mem_phi(Node *mem, BasicType ft, const Type *
   if (level <= 0) {
     return NULL; // Give up: phi tree too deep
   }
-  Node *start_mem = C->start()->proj_out_or_null(TypeFunc::Memory);
+  Node *start_mem = comp->start()->proj_out_or_null(TypeFunc::Memory);
   Node *alloc_mem = alloc->in(TypeFunc::Memory);
 
   uint length = mem->req();
   GrowableArray <Node *> values(length, length, NULL);
 
   // create a new Phi for the value
   PhiNode *phi = new PhiNode(mem->in(0), phi_type, NULL, mem->_idx, instance_id, alias_idx, offset);
-  transform_later(phi);
+  igvn->register_new_node_with_optimizer(phi);
```

Comment:
Will this result in a `transform_later` call on the `phi`?

Reply:
Yes, it will.